axelor-human-resource/src/main/java/com/axelor/apps/hr/db/repo/EmployeeHRRepository.java

```diff
@@ -24,39 +24,29 @@
 import com.axelor.apps.base.service.app.AppBaseService;
 import com.axelor.apps.hr.db.Employee;
 import com.axelor.apps.hr.db.EmploymentContract;
+import com.axelor.apps.hr.service.EmployeeComputeStatusService;
 import com.axelor.auth.AuthUtils;
 import com.axelor.auth.db.User;
 import com.axelor.inject.Beans;
 import com.google.common.base.Strings;
+import com.google.inject.Inject;
 import java.time.LocalDate;
 import java.util.Map;
 import java.util.Objects;
 import java.util.Optional;
 
 public class EmployeeHRRepository extends EmployeeRepository {
 
+  @Inject protected AppBaseService appBaseService;
+  @Inject protected EmployeeComputeStatusService employeeComputeStatusService;
```

Comment:
AppBaseService is not used in the repo, only in EmployeeComputeStatusService. Use Beans.get instead of injection.

Reply:
Can you please tell why Beans.get should be used instead of field injection ?
Is it due to any conventions ?
Thanks

Reply:
We generally want to use contructor injection and avoid Beans.get in services. But in repositories, since they are called in services, we want to avoid having injections in them. Preventing circulary injection. Also, like that we have a clearer (not optimal) structure :
Controller -> Services -> Repositories -> Beans.get and not Controller -> Services -> Repositories -> Services